wallet/api/client_check_transaction.go

```diff
@@ -0,0 +1,198 @@
+		h.interactor.Log(ctx, traceID, LogType(reportType), msg)
+	})
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, fmt.Errorf("could not find a healthy node: %w", err))
+		return nil, nodeCommunicationError(ErrNoHealthyNodeAvailable)
+	}
+
+	h.interactor.Log(ctx, traceID, InfoLog, "Retrieving latest block information...")
+	stats, err := currentNode.SpamStatistics(ctx, request.PubKey)
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, fmt.Errorf("could not get the latest block from the node: %w", err))
+		return nil, nodeCommunicationError(ErrCouldNotGetLastBlockInformation)
+	}
+	h.interactor.Log(ctx, traceID, SuccessLog, "Latest block information has been retrieved.")
+
+	if stats.LastBlockHeight == 0 {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, ErrCouldNotGetLastBlockInformation)
+		return nil, nodeCommunicationError(ErrCouldNotGetLastBlockInformation)
+	}
+
+	if stats.ChainID == "" {
+		h.interactor.NotifyError(ctx, traceID, NetworkError, ErrCouldNotGetChainIDFromNode)
+		return nil, nodeCommunicationError(ErrCouldNotGetChainIDFromNode)
+	}
+
+	err = h.spam.CheckSubmission(request, &stats)
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, ApplicationError, fmt.Errorf("could not send transaction: %w", err))
+		return nil, applicationCancellationError(ErrTransactionBlockedBySpamRules)
+	}
+
+	// Check the payload.
+	inputData, err := wcommands.ToMarshaledInputData(request, stats.LastBlockHeight)
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, InternalError, fmt.Errorf("could not marshal input data: %w", err))
+		return nil, internalError(ErrCouldNotCheckTransaction)
+	}
+
+	h.interactor.Log(ctx, traceID, InfoLog, "Checking the transaction...")
+	signature, err := w.SignTx(params.PublicKey, commands.BundleInputDataForChecking(inputData, stats.ChainID))
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, InternalError, fmt.Errorf("could not check the transaction: %w", err))
+		return nil, internalError(ErrCouldNotCheckTransaction)
+	}
+	h.interactor.Log(ctx, traceID, SuccessLog, "The transaction has been checked.")
+
+	// Build the transaction.
+	tx := commands.NewTransaction(params.PublicKey, inputData, &commandspb.Signature{
+		Value:   signature.Value,
+		Algo:    signature.Algo,
+		Version: signature.Version,
+	})
+
+	// Generate the proof of work for the transaction.
+	h.interactor.Log(ctx, traceID, InfoLog, "Computing proof-of-work...")
+	tx.Pow, err = h.spam.GenerateProofOfWork(params.PublicKey, &stats)
+	if err != nil {
+		h.interactor.NotifyError(ctx, traceID, InternalError, fmt.Errorf("could not compute the proof-of-work: %w", err))
+		return nil, internalError(ErrCouldNotCheckTransaction)
+	}
+	h.interactor.Log(ctx, traceID, SuccessLog, "The proof-of-work has been computed.")
+
+	h.interactor.NotifySuccessfulRequest(ctx, traceID, TransactionSuccessfullyChecked)
```

Comment:
The call to `currentNode.CheckTransaction()` is missing? Same in `admin_check_transaction.go`.

Reply:
This has been addressed earlier.
Thanks!